File: frontend/src/api/v2/sync.ts
import client from '../client';

export interface PipelineStartResponse {
  pipeline_run_id: number;
  status: string;
  phase: string;
  message: string;
  version?: string;
  generated_at?: string;
  request_id?: string;
}

export interface PipelineStatusResponse {
  pipeline_run_id: number;
  user_id: number;
  status: string;
  phase: string;
  last_error?: string | null;
  created_at?: string | null;
  started_at?: string | null;
  completed_at?: string | null;
  version?: string;
  generated_at?: string;
  request_id?: string;
}

export const startSyncPipelineV2 = async (params?: {
  mode?: 'incremental' | 'full';
  use_llm?: boolean;
  max_clusters?: number;
  min_clusters?: number;
}): Promise<PipelineStartResponse> => {
  const response = await client.post<PipelineStartResponse>('/v2/sync/start', null, {
    params: {
      mode: params?.mode ?? 'incremental',
      use_llm: params?.use_llm ?? true,
      max_clusters: params?.max_clusters ?? 8,
      min_clusters: params?.min_clusters ?? 3,
    },
  });
  return response.data;
};

export const getPipelineStatusV2 = async (
  runId: number
): Promise<PipelineStatusResponse> => {
  const response = await client.get<PipelineStatusResponse>(`/v2/sync/jobs/${runId}`);
  return response.data;
};

File: src/nebula/schemas/v2/sync.py
"""V2 sync pipeline schemas."""

from datetime import datetime

from pydantic import BaseModel, Field


class PipelineStartResponse(BaseModel):
    """Response when starting the v2 sync pipeline."""

    pipeline_run_id: int = Field(..., description="Pipeline run ID")
    status: str = Field(..., description="Initial pipeline status")
    phase: str = Field(..., description="Initial pipeline phase")
    message: str = Field(..., description="Human readable status")
    version: str | None = Field(None, description="Pipeline version token")
    generated_at: str | None = Field(None, description="Response generation timestamp")
    request_id: str | None = Field(None, description="Request correlation ID")


class PipelineStatusResponse(BaseModel):
    """Response for pipeline status query."""

    pipeline_run_id: int = Field(..., description="Pipeline run ID")
    user_id: int = Field(..., description="User ID")
    status: str = Field(..., description="Pipeline status")
    phase: str = Field(..., description="Current phase")
    last_error: str | None = Field(None, description="Latest error message")
    created_at: datetime | None = None
    started_at: datetime | None = None
    completed_at: datetime | None = None
    version: str | None = Field(None, description="Pipeline version token")
    generated_at: str | None = Field(None, description="Response generation timestamp")
    request_id: str | None = Field(None, description="Request correlation ID")
